### DISCUSSION TITLE
SimpleMLLPClient timeout error

### DISCUSSION THREAD

**Discussion by jorge-zapata-medstone** [nHapiNET/nHapi#417] [Q&A]
## Description:
Im having problems with the `SimpleMLLPClient` from the nhapiTools library i know this is not the same library but i was wondering if anyone has had any issues where the connection would timeout?.

## Code:
```csharp
try
{
    // Create connection object
    var connection = new SimpleMLLPClient(CONNECTION_URL, PORT_NUMBER, encoding);

    // Send the previously created HL7 message over the connection established
    IMessage responseMessage = connection.SendHL7Message(hl7Message);

    // Display the message response received from the remote party
    var responseString = pipeParser.Encode(responseMessage);

    return responseMessage;
}
catch (Exception e)
{
    throw new Exception($"Error occured while sending HL7 message {e.Message}");
}
```

In the example above if the `CONNECTION_URL` or the `PORT_NUMBER` is not available i never reach the `catch` portion of the code which leaves the program running indefinitely.

Does anyone know how to catch this error or avoid it?

## EDIT:
I just found out that i has a timeout of like 1-2 minutes is there a way i can reduce the timeout to idk 20-30 seconds?

**Comment by milkshakeuk** [comment]
@dib0 is this something you can help with?

@jorge-zapata-medstone, @dib0 is the author of the nhapitools library.

**Comment by dib0** [comment]
Hi,

TCP connections can timeout. Most systems should keep the TCP connection for HL7 messages open, so a timeout shouldn't happen too often. If it does, please check with the vendor or specs of the other system.

You can decrease the timeout when calling the constructor. The default setting is 30 seconds (check below).
`public SimpleMLLPClient(string hostname, int port, int receiveTimeout = 30000)`

You can set it to whatever timeout you want (it is in milliseconds). For example (10 seconds):
`var client = new SimpleMLLPClient(192.168.0.5, 1000, 10000)`

On a timeout the method should send a TimeoutException, do your receive those?

Hope this helps!

BTW the github repo for NHapiTools is over here:
https://github.com/dib0/NHapiTools
